### PR TITLE
Re-enable cudagraphs for benchmark scripts

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94192

Related to https://github.com/pytorch/pytorch/pull/93253

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by albanD — APPROVED** [review]

**Comment by weiwangmeta** [comment]
https://github.com/pytorch/pytorch/actions/runs/4106055000/jobs/7086779397#step:11:1273 this error was seen multiple times in the past with some of the previous commits. I have started the 3rd attempt: https://github.com/pytorch/pytorch/actions/runs/4106055000

**Comment by weiwangmeta** [comment]
https://github.com/pytorch/pytorch/pull/94088 tracked that recently the model's performance speedup is always above 1.20x, so I bumped up threshold from 1.185x to 1.20x. It seems this PR is giving a speedup value of 1.1938x https://github.com/pytorch/pytorch/actions/runs/4110033172/jobs/7092706171#step:11:1140. 

Though I'm not sure what caused the performance speedup to increase from 1.185x-1.205x range to 1.20x-1.22x range. If you think this is normal speedup, please change this (https://github.com/pytorch/pytorch/blob/579ae64d81b7c14313804ca82716dc52f07ce18a/benchmarks/dynamo/check_hf_bert_perf_csv.py#L19)  to a smaller number that you think it would indeed be a regression. 

cc @ngimel @desertfire

**Comment by jansel** [comment]
Thanks, I'll set it 1.19x.  This is more changing what is measured than changing speedups.

**Comment by desertfire** [comment]
The `OPTForCausalLM` OOM might be a real regression that happened in the past a few days which wasn't caught because CI were testing non-cudagraph version. Okay to skip so CI can run the cudagraph version.

**Comment by jansel** [comment]
> The `OPTForCausalLM` OOM might be a real regression that happened in the past a few days which wasn't caught because CI were testing non-cudagraph version. Okay to skip so CI can run the cudagraph version.

I'll add a skip for it.

File: benchmarks/dynamo/huggingface.py [hunk 1]
```diff
@@ -98,6 +98,8 @@ def pip_install(package):
     # Fails with even batch size = 1
     "GPTJForCausalLM",
     "GPTJForQuestionAnswering",
+    # Out of memory in CI
+    "OPTForCausalLM",
```
Comment (desertfire) [inline benchmarks/dynamo/huggingface.py:102]: Better to add to this list if this is a CI-only failure, https://github.com/pytorch/pytorch/blob/333e771394998e6eb1d8b451cae22717123c07fc/benchmarks/dynamo/common.py#L148

**Review by desertfire — CHANGES_REQUESTED** [review]
```
python benchmarks/dynamo/huggingface.py --ci --training --accuracy --device cuda --backend inductor --amp --only OPTForCausalLM
cuda train OPTForCausalLM                      [2023-02-08 01:30:14,709] torch._inductor.compile_fx: [WARNING] skipping cudagraphs due to input mutation
```
This is what I got locally, "skipping cudagraphs", not sure what it going on CI...

File: benchmarks/dynamo/huggingface.py (hunk 1 above)
Comment (jansel) [reply benchmarks/dynamo/huggingface.py]: Changed

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**